app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -99,8 +100,23 @@
       });
     };
 
-    var updateEnvFromEntries = function(entries) {
-      ctrl.envFromEntries = entries || [];
+    var getReferenceValues = function(option) {
```

Comment:
You might be able to nix the switch statements & eliminate some duplication with:

```JavaScript
var getReferenceValues = function(option) {
      var kindRef = _.camelCase(option.kind) + 'Ref';
      return _.filter(ctrl.envFromEntries, [kindRef, 'name', option.metadata.name]);      
    };
```
(untested)

Reply:
Prob similar could be done to trim down [lines 76-91](https://github.com/openshift/origin-web-console/pull/2238/files#diff-a09b8976dc42ea9f1ff076ac68341ba9R78)

Reply:
@benjaminapetersen looks like coincidentally wiped some of the duplication out. Good catch on the switch